### DISCUSSION TITLE
Reset PVoutput upload flags and reupload

### DISCUSSION THREAD

**Discussion by loudneighbor** [SBFspot/SBFspot#492] [Q&A] [answered]
Hi,
Newbie to SBFspot and PVoutput and I'd like to understand the process to clear any/all uploads to PVoutput in the past, say, 30 days;  further, to reupload the records set to zero.

My first assumption - wrong? is that resetting the PVoutput field to 0 in DayData flags those records as not yet uploaded.
Is this correct?

If not, please ignore the rest.
If correct, please continue...

Note #### is the numeric value for TimeStamp from 30 days ago

UPDATE DayData
SET PVoutput> = <0> 
WHERE​ PVoutput=1 TimeStamp>####; 

Once these flags are reset, what, if anything, must I do to ensure the records are uploaded?

Will the daemon automatically scan the "not yet uploaded" records from DayData or must I do more?

Should I delete all uploaded values on PVOutput or will any new uploads replace what is there?

Will th

**Comment by sbf-** [comment, marked as answer]
Almost correct ;-)
PVoutput flag should be 'null' instead of '0'

To reset the records for today:
``` SQL
update daydata
set PVoutput=NULL 
where date(TimeStamp, 'unixepoch', 'localtime') = '2021-10-07';
```
But I recommend to select the records first for verification:
``` SQL
select * from daydata
where date(TimeStamp, 'unixepoch', 'localtime') = '2021-10-07';
```
The upload daemon will resend the records, no need to remove them at pvoutput.org

Be aware that uploads to PVO are limited to 14 days (or 90 days in donation mode)

**Comment by loudneighbor** [comment]
Thanks, much.  That did the trick VERY NICELY.  And I like that timestamp trick to get "real" format for a date.

Before I cut you loose - or vice versa - do you happen to know how long Sunny Boy inverter keeps data?  The unit is less than 2 years old.

And any tricks for pulling any historical data from it?  

BTW, I donated to PVoutput before I configured the Pi and before I bought the Eagle.
Enjoying it much - and learning much.

-Alex

**Reply by sbf-** [reply]
> And any tricks for pulling any historical data from it?

Daydata (every 5mins) is kept for aprox 2 months. To get them all use `-ad62` (Archive Days)
Monthdata/events are kept almost forever. In your case 2 years or 24 months `-am24` (Archive Months) and `-ae24` (Archive Events)
Spotdata is not archived.
`sbfspot -?` for all secrets

Until now, daemon is not sending monthdata to PVO, but it can be done manually with [the CSV loader](https://pvoutput.org/load.jsp)
